### PR TITLE
[ci_runner] Clean up deprecated BazelCommands syntax

### PR DESCRIPTION
**Pull request by maggie-lou**
`BazelCommands` was replaced with `Steps`, so that we can support arbitrary bash code in our remote runners.

### CODE REVIEW

File: enterprise/server/cmd/ci_runner/main.go [hunk 1]
```diff
@@ -1256,164 +1250,6 @@ func (ar *actionRunner) Run(ctx context.Context, ws *workspace) error {
 		}
 	}
```
Comment (bduffany) [inline enterprise/server/cmd/ci_runner/main.go:1252]: wooo nice :)

File: enterprise/server/test/integration/remote_bazel/remote_bazel_test.go [hunk 2]
```diff
@@ -498,3 +480,129 @@ func TestBuildRemotelyRunLocally(t *testing.T) {
 	require.NoError(t, err)
 	require.NotContains(t, string(logResp.GetBuffer()), "Hello! I'm a go program.")
 }
+
+func TestAccessingSecrets(t *testing.T) {
```
Comment (bduffany) [inline enterprise/server/test/integration/remote_bazel/remote_bazel_test.go:484]: is this change intentional?

File: docs/workflows-config.md [hunk 3]
```diff
@@ -47,44 +47,43 @@ actions:
       pull_request:
         branches:
           - "*"
-    bazel_commands:
-      - "test //..."
+    steps:
```
[the remark this answers is not in the document]
Comment (maggie-lou) [reply docs/workflows-config.md:50]: Hmm personally I'd rather wait until we have a clearer idea for what we want for the UI, before implying the feature is coming in our docs and drawing attention to it. Customers should be incentivized to set a name later if they care about having readable titles. I don't feel like the UI feature request is top priority for now

I was imagining that this PR should feel like a backend refactor and customers should ideally not even realize it's happening

[the remark this answers is not in the document]
Comment (maggie-lou) [reply docs/workflows-config.md:50]: Good catch - fixed

[the remark this answers is not in the document]
Comment (bduffany) [reply docs/workflows-config.md:50]: Fair enough - if we do add the collapsible sections later, we can probably just add some default names like "Step 1", "Step 2" etc.

**Review by bduffany — APPROVED** [review]
